deployments/cardano-nodes-config.nix

```diff
@@ -1,12 +1,160 @@
+                                      [ "allow-deployer-ssh-${region}"
+                                        "allow-kademlia-public-udp-${region}"
+                                        "allow-cardano-public-tcp-${region}" ]
+                                    else throw "While computing EC2 SGs: unhandled cardano-node type: '${type}', must be either 'core' or 'relay'";
+                             }; } ) topologyList;
+    byName        = name: let xs = filter (n: n.name == name) indexed;
+                          in if xs != [] then builtins.elemAt xs 0
+                             else throw "No indexedNode by name '${name}'.";
+    # Canonical node parameter format is:
+    #   i             :: Int
+    #   region        :: String
+    #   type          :: String               -- one of: 'core', 'relay'
+    #   static-routes :: [['nodeId, 'nodeId]] -- here we go, TupleList..
+    #
+    # WARNING: this depends on the sort order, as explained above.
+    firstRelay      = findFirst (x: x.value.type == "relay") (throw "No relays in topology, it's sad we can't live..") indexed;
+    nRelays         = length relays;
+    firstRelayIndex = firstRelay.value.i;
+    allNames        = map (x: x.name) indexed;
+    cores           = filter (x: x.value.type == "core")  indexed;
+    relays          = filter (x: x.value.type == "relay") indexed;
+    canonical       = builtins.listToAttrs (cores ++ relays);
+    ##
+    ##
+    regionSGNames = region:
+        [ "allow-deployer-ssh-${region}"
+          "allow-kademlia-public-udp-${region}"
+          "allow-cardano-public-tcp-${region}"
+        ];
+    regionSGs      = nodePort: region: {
+        "allow-deployer-ssh-${region}" = {
+          inherit region accessKeyId;
+          description = "SSH";
+          rules = [{
+            protocol = "tcp"; # TCP
+            fromPort = 22; toPort = 22;
+            sourceIp = deployerIP + "/32";
+          }];
+        };
+        "allow-kademlia-public-udp-${region}" = {
+          inherit region accessKeyId;
+          description = "Kademlia UDP public";
+          rules = [{
+            protocol = "udp"; # UDP
+            sourceIp = "0.0.0.0/0";
+            fromPort = 1024; toPort = 65535;
+          }];
+        };
+        "allow-cardano-public-tcp-${region}" = {
+          inherit region accessKeyId;
+          description = "Cardano TCP public (+WWW)";
+          rules = [{
+            protocol = "tcp"; # TCP
+            fromPort = nodePort; toPort = nodePort;
+            sourceIp = "0.0.0.0/0";
+          }{
+            protocol = "tcp";
+            fromPort = 8080; toPort = 8080;
+            sourceIp = "0.0.0.0/0";
+          }];
+        };
+      };
+    coreSGNames = core:
+        [ "allow-cardano-static-peers-${core.name}-${core.value.region}" ];
+    coreSGs     = nodePort: ips: core:
+      let neighbourNames = traceSF (p: "${core.name} peers: " + concatStringsSep ", " core.value.peers) core.value.peers;
```

Comment:
No use of traceSF by default, it creeps the output

Reply:
@domenkozar, it's arguably critical information -- we're now having more steps in the pipeline than we had before -- and it's getting worse.

Having visibility helps..

But if you're feeling strongly about it, of course we can dump it.

Reply:
Got annoyed by these traces as well, so dropped them
.